### PR TITLE
Modernize Kupfer Look and Feel

### PR DESCRIPTION
**Pull request by chaitanya-lakkundi**
I am impressed by the simplicity of Kupfer launcher and have done the following changes to improve its look:

1. Changed the look and feel of Kupfer
new kupfer
kupfer firefox
2. Added --preferences CLI option
3. Migrated the CSS code into a template file and integrated with Gnome Color Chooser to generate the actual style.css file ( which is used in browser.py).
4. Added --colorize CLI option. Kupfer browser window colour can now be changed . (Gnome Color Chooser)
kupfer colorize

What can be done ?
1. This new look can be added as an option along with the look existing now.

### CODE REVIEW

**Comment by bluss** [comment]
Cool but, can we somehow make sure that only css changes are needed? That would be ideal. Of course commits that make css styling more flexible, without changing the default style, are welcome.

**Comment by bluss** [comment]
I would hold off on the --preferences change as it is now. It needs to take single instance into account.

File: kupfer.py [hunk 1]
```diff
@@ -19,4 +20,7 @@
 
 if __name__ == '__main__':
     from kupfer import main
+#    from os import chdir
+#    chdir("/home/chaitanya/Documents/PROJECTS/kupfer/")
+#    chdir is required if kupfer script added in OS starup (start on boot). Otherwise all the applications are not indexed.
```
Comment (bluss) [inline kupfer.py]: This sounds like a bug

File: kupfer.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/python3
```
Comment (bluss) [inline kupfer.py]: This seems unneeded.

File: kupfer/ui/style.css [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#kupfer {
```
Comment (bluss) [inline kupfer/ui/style.css]: Super cool, but I will put in my caution: Css styling of GTK is messy, and it seems to change with every version of GTK. This default style file doesn't look like it defines enough colors and properties, so the style will be a mix of this file and the current theme, which can turn out very ugly.

File: kupfer.py (hunk 1 above)
Comment (chaitanya-lakkundi) [inline kupfer.py]: I use Arch Linux. May be a bug with that..
But always I have observed that whenever I start Kupfer **after booting**, all the applications are **correctly indexed**. This is not the case when Kupfer is scheduled for autostart.

File: kupfer.py (hunk 2 above)
Comment (chaitanya-lakkundi) [inline kupfer.py]: I included this line only to facilitate application autostart (after booting).

File: kupfer/ui/style.css (hunk 3 above)
Comment (chaitanya-lakkundi) [inline kupfer/ui/style.css]: The current scheme of styling uses only CSS. What could be another alternative to facilitate window styling ? But I should say that CSS styling is easier .

**Comment by chaitanya-lakkundi** [comment]
What kind of change should I inculcate in my PR to add a new theme besides existing theme (without modifying the current layout) ?

File: kupfer.py (hunk 1 above)
Comment (bluss) [inline kupfer.py]: Report a bug for that, be specific about what correctly indexed means (expected and actual behaviour). Linux is very diverse so there are a lot of cases.

File: kupfer.py (hunk 2 above)
Comment (bluss) [inline kupfer.py]: Then it's a separate bug or not correctly installed.

**Comment by bluss** [comment]
I don't have the answer, CSS styling is new to me. I've asked around, and we can't change Gtk.Box orientation (V/H) from css alone.

A. Use a new style property. Python bindings don't seem to implement bindings for this, so it's not possible
B. Use separate mechanism to toggle horizontal/vertical mode in Kupfer. A single toggle should be all, as much as possible should be in CSS.

**Comment by khurshid-alam** [comment]
I appreciate css styling, But first it should be done without changing the current layout.

**Comment by chaitanya-lakkundi** [comment]
I will add a toggle which clearly would state change of layout and create another pull request. Separate PR only for CSS styling will also be made.

**Comment by khurshid-alam** [comment]
@chaitanya-lakkundi Thanks. I look forward to that.

**Comment by khurshid-alam** [comment]
@chaitanya-lakkundi We need your help. Please refer to this issue #66. Thanks.

**Pull request closed without merging**